contracts/PriceFeed.sol

```diff
@@ -75,147 +82,239 @@ contract PriceFeed is IPriceFeed {
 
   /**
    * @notice Get the time-weighted quote of `quoteToken` received in exchange for a `baseAmount`
-   * of `baseToken`, from the pool with highest liquidity, based on a `secondsAgo` twap interval.
+   * of `baseToken`, from the pool with highest liquidity, based on a `secondsTwapInterval` twap interval.
    * @param baseAmount Amount of baseToken to be converted
    * @param baseToken Address of an ERC20 token contract used as the baseAmount denomination
    * @param quoteToken Address of an ERC20 token contract used as the quoteAmount denomination
-   * @param secondsAgo Number of seconds in the past from which to calculate the time-weighted quote
+   * @param secondsTwapInterval Number of seconds in the past from which to calculate the time-weighted quote
    * @return quoteAmount Equivalent amount of ERC20 token for baseAmount
    *
-   * Requirement: `secondsAgo` must be greater than 0.
    * Note: If a pool does not exist or a valid quote is not returned execution will not revert and
    * `quoteAmount` will be 0.
    */
   function getQuote(
     uint128 baseAmount,
     address baseToken,
     address quoteToken,
-    uint32 secondsAgo
+    uint32 secondsTwapInterval
   ) public view returns (uint256 quoteAmount) {
     address pool = getPool(baseToken, quoteToken).poolAddress;
 
     if (pool != address(0)) {
-      int24 arithmeticMeanTick = _getArithmeticMeanTick(pool, secondsAgo);
-
-      quoteAmount = OracleLibrary.getQuoteAtTick(
-        arithmeticMeanTick,
-        baseAmount,
-        baseToken,
-        quoteToken
-      );
+      // Get spot price
+      if (secondsTwapInterval == 0) {
+        // Get sqrtPriceX96 from slot0
+        (uint160 sqrtPriceX96, , , , , , ) = IUniswapV3Pool(pool).slot0();
+        quoteAmount = _getQuoteAtSqrtPriceX96(sqrtPriceX96, baseAmount, baseToken, quoteToken);
```

Comment:
This returns the *price* (and the amount you'd for a given tick), not a real *quote* -> this price is only valid within a tick, if the amountIn is enough, price might change (ie liquidity from the tick is consumed and the next non-empty tick is used) resulting in a smaller amount out

-> To get a real quote, there is no other choice than simulating an actual swap (ie try-catch calling a contract swapping and reverting while returning the amount received) -> quite costly if onchain tho:/

Reply:
If with _quote_ you mean the amount that you'd get by swapping, yes that's how it should be. The price feed is designed to return the absolute amount for a currency pair, and should be unrelated to swaps.

You can also find this in the gotchas FAQs
> Quotes represent a time-weighted average price for a currency in a certain amount of time (see [TWAP oracles](https://docs.uniswap.org/protocol/concepts/V3-overview/oracle)). As such they don't exactly correspond to the amount displayed during a swap on Uniswap, but they're better suited to estimate the value of a currency.

Reply:
@jjranalli I meant the spot price (used to get a quote), not the twap;)

Reply:
This is a correct spot price but cannot be used to get a quote (ie an amount out), as crossing ticks changes it

Reply:
> I meant the spot price

true. Should modify the gotcha to be more generic and explicitly refer to spot price too.

> spot price used to get the quote, which is "how many token out do I get for amount in?"

Thing is, the price feed quote is "how many `quoteToken`s is a `baseAmount` of `baseToken`s worth". This is independent of liquidity / inAmount, and doesn't aim to estimate a potential swap. Does it make sense?

Reply:
closing readme gotcha has been updated